internal/pkg/deploy/cloudformation/stack/scheduled_job.go

```diff
@@ -0,0 +1,315 @@
+	}
+}
+
+func awsCronFieldSpecified(input string) bool {
+	return !strings.ContainsAny(input, "*?")
+}
+
+// toAWSCron converts "standard" 5-element crons into the AWS preferred syntax
+// cron(* * * * ? *)
+// MIN HOU DOM MON DOW YEA
+// EITHER DOM or DOW must be specified as ? (either-or operator)
+// BOTH DOM and DOW cannot be specified
+// DOW numbers run 1-7, not 0-6
+// Example input: 0 9 * * 1-5 (at 9 am, Monday-Friday)
+//              : cron(0 9 ? * 2-6 *) (adds required ? operator, increments DOW to 1-index, adds year)
+func toAWSCron(schedule string) (string, error) {
+	const (
+		MIN = iota
+		HOU
+		DOM
+		MON
+		DOW
+	)
+
+	// Split the cron into its components. We can do this because it'll already have been validated.
+	// Use https://golang.org/pkg/strings/#Fields since it handles consecutive whitespace.
+	sched := strings.Fields(schedule)
+
+	// Check whether the Day of Week and Day of Month fields have a ?
+	// Possible conversion:
+	// * * * * * ==> * * * * ?
+	// 0 9 * * 1 ==> 0 9 ? * 1
+	// 0 9 1 * * ==> 0 9 1 * ?
+	switch {
+	// If both are unspecified, convert DOW to a ? and DOM to *
+	case !awsCronFieldSpecified(sched[DOM]) && !awsCronFieldSpecified(sched[DOW]):
+		sched[DOW] = "?"
+		sched[DOM] = "*"
+	// If DOM is * or ? and DOW is specified, convert DOM to ?
+	case !awsCronFieldSpecified(sched[DOM]) && awsCronFieldSpecified(sched[DOW]):
+		sched[DOM] = "?"
+	// If DOW is * or ? and DOM is specified, convert DOW to ?
+	case !awsCronFieldSpecified(sched[DOW]) && awsCronFieldSpecified(sched[DOM]):
+		sched[DOW] = "?"
+	// Error if both DOM and DOW are specified
+	default:
+		return "", errors.New("cannot specify both DOW and DOM in cron expression")
+	}
+
+	// Increment the DOW by one if specified as a number
+	// https://play.golang.org/p/_1uxt0zJneb
+	var newDOW []rune
+	for _, c := range sched[DOW] {
+		if unicode.IsDigit(c) {
+			newDOW = append(newDOW, c+1)
+		} else {
+			newDOW = append(newDOW, c)
+		}
+	}
+	// We don't need to use a string builder here because this will only ever have a max of
+	// about 50 characters (SUN-MON,MON-TUE,TUE-WED,... is the longest possible string here)
+	sched[DOW] = string(newDOW)
+
+	// Add "every year" to 5-element crons to comply with AWS
+	sched = append(sched, "*")
+
+	return fmt.Sprintf(fmtCronScheduleExpression, strings.Join(sched, " ")), nil
+}
+
+// StateMachine converts the Timeout and Retries fields to an instance of template.StateMachineOpts
+// It also performs basic validations to provide a fast feedback loop to the customer.
+func (j *ScheduledJob) stateMachineOpts() (*template.StateMachineOpts, error) {
+	var timeoutSeconds *int
+	if j.manifest.Timeout != "" {
+		parsedTimeout, err := time.ParseDuration(j.manifest.Timeout)
+		if err != nil {
+			return nil, err
+		}
+		if parsedTimeout < 1*time.Second {
+			return nil, errors.New("timeout must be greater than or equal to 1 second")
+		}
+		if parsedTimeout.Seconds() != math.Round(parsedTimeout.Seconds()) {
```

Comment:
What do you think of this?
```suggestion
		if parsedTimeout.Seconds() != parsedTimeout.Truncate(time.Second) {
```